# Seattle Kids in Urban Centers

## Problem Statement:

## Background
Welcome to Seattle, the big city with a small town feeling.  Though Seattle is by all acounts a major metropolitan area, the small neighborhood focus of the city lends to a small town feeling.  Across the city there are 32 'urban villages' which act as mini-downtowns.  These higher-density urban areas within the city are not accidental but the reuslt of an approach to focus development in specific areas of the city.  The areas targeted for development are typically along major corridors with transportation.  This approach avoids rezoning large fractions of residential land currently slated for single-family housing only.  (add citation) Today, most of the residential land in the city remains zoned exclusively for single-family homes and most developable land in these areas has been used.  (add citation)  

In July 2019, the city enacted legislation allowing the construction of accessory dwelling units (ADUs) on single-family zoned property. (add citation) This change was enacted to help with the severe housing affordability crisis the city was facing.  While ADUs were constructed in record numbers (add citation) the housing crisis continues and the city is currently examining zoning restrictions in all areas of the city on a neighborhood-by-neighborhood basis.  This rezoning has focused on the urban centers and villages 

The city of Seattle has seen immense population growth over the last twenty years.  While most residential land in the city has been zoned for single-family housing the explosive growth of the last two decades has resulted in significant growth in the more urban areas of the city.  The city designates these areas as 32 individual 'Urban Centers and Villages' (UCV).  The UCVs are subject to different zoning than the surrounding neighborhood and are considered targets for dense development including multi-family (e.g, high-rise apartments) as well as mixed-used residential/commercial.  

Enrollment Seattle Public Schools (SPS) has seen a lot of changes over the same time period.  The relative young average age combined with millenials generally delaying life milestones compared to previous generations led to a decline in per capita children population during the early 2000s.  (add citation) But the millenial generation is aging and the number of children in the city is rising significantly.  In 2019, in response to the rising student population, SPS reopened a high school on the north end that had been closed for 38 years. With rapidly changing demographics, the city has been slow to respond to the growing need for schools in more urban parts of the city.  This includes the high density areas in South Seattle as well as the downtown core.  The prevailing attitude is that "people don't have children in apartments" or more explictly, "people don't want to have children in apartments."  The experiences of families living in multifamily housing have been summarily dismissed as unpalatable.   



## Data Collection

Population data was retrieved from the city of Seattle [population data website](https://www.seattle.gov/opcd/population-and-demographics/decennial-census) on December 5, 2021. These reports are based on early results from the 2020 census, aggregated by city-designated urban centers (subset of the city) as well as by larger neighborhood areas (covering entire city).  

Enrollment data was retrieved from [Seattle Public Schools](https://www.seattleschools.org/departments/enrollment-planning/enrollment-data/annual-enrollment-reports/) on December 5, 2021.  The enrollment data 

In [1]:
#imports
import pandas as pd
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt

import dash
from dash import dcc
from dash import html
import plotly.express as px

## Population Data

In [2]:
# get population data
ucvpop = pd.read_csv("./data/citydata/Urban_Centers_and_Villages_with_PL_94-171_Redistricting__Data_for_1990-2020.csv")
ucvpop.set_index('OBJECTID', inplace = True)
ucvpop.head()

NEIGH_NO               NEIGH_NAME UV_TYPE  \
OBJECTID                                              
1              106         Upper Queen Anne     RUV   
2              302                  Fremont     HUV   
3              205    West Seattle Junction     HUV   
4              111            Columbia City     RUV   
5              414  First Hill/Capitol Hill      UC   

                          TYPE_NAME  AREA_ACRES  F1990_PL_data_TOT_POP  \
OBJECTID                                                                 
1         Residential Urban Village   52.658720                   1921   
2                 Hub Urban Village  214.511083                   3153   
3                 Hub Urban Village  270.417026                   3627   
4         Residential Urban Village  334.543340                   4281   
5                      Urban Center  916.221801                  28940   

          F1990_PL_data_WHITE_NOT  F1990_PL_data_BLACK_NOT  \
OBJECTID                                                     
1                            1715                       56   
2                            2697                       90   
3                            3169                      145   
4                             898                     1827   
5                           21504                     3037   

          F1990_PL_data_AMI_NOT  F1990_PL_data_ASIAN_NOT  ...  \
OBJECTID                                                  ...   
1                            11                       71  ...   
2                            64                      189  ...   
3                            33                      150  ...   
4                           111                     1269  ...   
5                           534                     2403  ...   

          F1990_PL_data_GQ_I  F1990_PL_data_GQ_N_I  F2000_PL_data_GQ_I  \
OBJECTID                                                                 
1                          0                    37                   0   
2                          0                    47                   0   
3                        401                    14                 358   
4                          0                     5                  98   
5                        660                  1370                 782   

          F2000_PL_data_GQ_N_I  F2010_PL_data_GQ_I  F2010_PL_data_GQ_N_I  \
OBJECTID                                                                   
1                            0                   0                     4   
2                           63                   0                    90   
3                           13                 272                     0   
4                            7                  95                    20   
5                         2278                 332                  2329   

          F2020_PL_data_GQ_I  F2020_PL_data_GQ_N_I   Shape__Area  \
OBJECTID                                                           
1                          0                     0  2.293805e+06   
2                          0                    66  9.344065e+06   
3                        235                    12  1.177932e+07   
4                        102                    80  1.457265e+07   
5                        846                  3135  3.991046e+07   

          Shape__Length  
OBJECTID                 
1           11354.22008  
2           16236.22258  
3           20674.93070  
4           25210.67846  
5           31324.33554  

[5 rows x 110 columns]

In [3]:
print('UCV data shape:', ucvpop.shape)

UCV data shape: (32, 110)


In [4]:
print('nulls in UCV population data:', ucvpop.isna().sum().sum())

nulls in UCV population data: 0


The UCV data has no nulls.  We have 110 columns for each of the 32 Urban Centers/Villages

In [5]:
# get population data
pop = pd.read_csv('./data/citydata/Community_Reporting_Areas_with_PL_94-171_Redistricting__Data_for_1990-2020.csv')
pop.set_index('OBJECTID', inplace= True)
pop.head()

NEIGH_NO                          NEIGH_NAME  \
OBJECTID                                                 
1              1.1                       Arbor Heights   
2              1.2                  Fauntleroy/Seaview   
3              1.3  West Seattle Junction/Genesee Hill   
4              1.4                        Alki/Admiral   
5              2.1                      North Delridge   

                                                 DETL_NAMES  NEIGHDIST_NO  \
OBJECTID                                                                    
1              Arbor Heights, Brace Point, Endolyne, Arroyo             1   
2         Fauntleroy, Gatewood, Morgan Junction, Lincoln...             1   
3         Seaview, Fairmount Park, Morgan Junction, Gene...             1   
4                                             Alki, Admiral             1   
5         North Delridge, Pigeon Point, Avalon, Luna Par...             2   

                  NEIGHDIST_NAME  AREA_ACRES  AREA_SQMI  \
OBJECTID                                                  
1                      Southwest       785.9       1.23   
2                      Southwest      1393.1       2.18   
3                      Southwest      1447.5       2.26   
4                      Southwest       898.3       1.40   
5         Delridge Neighborhoods      1606.9       2.51   

          F1990_PL_data_TOT_POP  F1990_PL_data_WHITE_NOT  \
OBJECTID                                                   
1                          6247                     5603   
2                         12919                    11746   
3                         16017                    14607   
4                          9880                     9212   
5                          3724                     2844   

          F1990_PL_data_BLACK_NOT  ...  F1990_PL_data_GQ_I  \
OBJECTID                           ...                       
1                              79  ...                   0   
2                             261  ...                 122   
3                             249  ...                 504   
4                              82  ...                 143   
5                             315  ...                   0   

          F1990_PL_data_GQ_N_I  F2000_PL_data_GQ_I  F2000_PL_data_GQ_N_I  \
OBJECTID                                                                   
1                            0                   0                     8   
2                           23                 103                     9   
3                            0                 431                     0   
4                            3                 101                     0   
5                           84                   0                    49   

          F2010_PL_data_GQ_I  F2010_PL_data_GQ_N_I  F2020_PL_data_GQ_I  \
OBJECTID                                                                 
1                          5                     3                   0   
2                         86                    25                  53   
3                        350                    20                 299   
4                        107                    24                 149   
5                          1                    16                   0   

          F2020_PL_data_GQ_N_I   Shape__Area  Shape__Length  
OBJECTID                                                     
1                            5  3.423387e+07   27702.727646  
2                           48  6.068510e+07   48830.048317  
3                           94  6.305462e+07   38739.896257  
4                           33  3.912974e+07   50339.656415  
5                           52  6.999494e+07   64237.742682  

[5 rows x 112 columns]

In [6]:
print('citywide population data shape:',pop.shape)

citywide population data shape: (53, 112)


In [7]:
print('nulls in citywide population data:',pop.isna().sum().sum())

nulls in citywide population data: 0


The broader neighborhood data has no nulls.  It is broken down into 53 neighborhoods spanning the entire city with 112 data columns for each neighborhood.

## First look at overall population changes
Adding a `change` feature that represents the change in population from 2010 to 2020 and a `perc_change` that represents the percentage change in population from 2010 to 2020

In [8]:
ucvpop['change'] = ucvpop['F2020_PL_data_TOT_POP'] - ucvpop['F2010_PL_data_TOT_POP']
pop['change'] = pop['F2020_PL_data_TOT_POP'] - pop['F2010_PL_data_TOT_POP']

In [9]:
ucvpop['perc_change'] = (ucvpop['F2020_PL_data_TOT_POP'] - ucvpop['F2010_PL_data_TOT_POP'])/ucvpop['F2010_PL_data_TOT_POP']
pop['perc_change'] = (pop['F2020_PL_data_TOT_POP'] - pop['F2010_PL_data_TOT_POP'])/pop['F2010_PL_data_TOT_POP']

In [10]:
ucvpop[['NEIGH_NAME', 'AREA_ACRES','F2010_PL_data_TOT_POP','F2020_PL_data_TOT_POP', 'change', 'perc_change']].sort_values('perc_change', ascending=False).head()

NEIGH_NAME   AREA_ACRES  F2010_PL_data_TOT_POP  \
OBJECTID                                                              
7              South Lake Union   339.886927                   3774   
3         West Seattle Junction   270.417026                   5043   
6                     Roosevelt   170.211776                   2616   
12             Greater Duwamish  4953.281793                   1064   
4                 Columbia City   334.543340                   4609   

          F2020_PL_data_TOT_POP  change  perc_change  
OBJECTID                                              
7                         12905    9131     2.419449  
3                          9160    4117     0.816379  
6                          4678    2062     0.788226  
12                         1836     772     0.725564  
4                          7661    3052     0.662183

In [11]:
pop[['NEIGH_NAME', 'AREA_ACRES','F2010_PL_data_TOT_POP','F2020_PL_data_TOT_POP', 'change', 'perc_change']].sort_values('perc_change', ascending=False).head()

NEIGH_NAME  AREA_ACRES  F2010_PL_data_TOT_POP  \
OBJECTID                                                                
47                Cascade/Eastlake       727.6                  13526   
44                         Ballard       492.9                   6739   
12                   Duwamish/SODO      2343.5                   2354   
21                    Judkins Park       202.9                   2925   
52        Downtown Commercial Core       206.5                   4070   

          F2020_PL_data_TOT_POP  change  perc_change  
OBJECTID                                              
47                        29182   15656     1.157474  
44                        11191    4452     0.660632  
12                         3611    1257     0.533985  
21                         4284    1359     0.464615  
52                         5813    1743     0.428256

In [12]:
print('UCV population:',sum(ucvpop['F2020_PL_data_TOT_POP']))
print('Total population:',sum(pop['F2020_PL_data_TOT_POP']))
print('UCV population fraction', round(sum(ucvpop['F2020_PL_data_TOT_POP'])/sum(pop['F2020_PL_data_TOT_POP']),3))

UCV population: 308307
Total population: 737015
UCV population fraction 0.418


In [13]:
print('UCV area (acres):',sum(ucvpop['AREA_ACRES']))
print('Total area (acres):',sum(pop['AREA_ACRES']))
print('UCV area fraction', round(sum(ucvpop['AREA_ACRES'])/sum(pop['AREA_ACRES']),3))

UCV area (acres): 16016.586737009999
Total area (acres): 53560.99999999999
UCV area fraction 0.299


In 2020, approximately 41.8% of the city population lived in an urban center or village while the urban villages only accounted for 29.9% of the city acreage.

In [14]:
# growth 2010-2020
print('All-ages population growth from 2010 to 2020')
print('UCV population growth:',(sum(ucvpop['F2020_PL_data_TOT_POP'])-sum(ucvpop['F2010_PL_data_TOT_POP'])))
print('Overall population growth:',(sum(pop['F2020_PL_data_TOT_POP'])-sum(pop['F2010_PL_data_TOT_POP'])))
print()
print('All-ages population growth rate from 2010 to 2020')
print('UCV population growth:',(sum(ucvpop['F2020_PL_data_TOT_POP'])-sum(ucvpop['F2010_PL_data_TOT_POP']))/sum(ucvpop['F2010_PL_data_TOT_POP']))
print('Overall population growth:',(sum(pop['F2020_PL_data_TOT_POP'])-sum(pop['F2010_PL_data_TOT_POP']))/sum(pop['F2010_PL_data_TOT_POP']))

All-ages population growth from 2010 to 2020
UCV population growth: 87281
Overall population growth: 128355

All-ages population growth rate from 2010 to 2020
UCV population growth: 0.3948901939138382
Overall population growth: 0.21088128018926822


## Kids

In [15]:
print('\nKids in UCVs\n')
print('2010 ',sum(ucvpop['F2010_PL_data_POP_UNDER_18']))
print('2020 ',sum(ucvpop['F2020_PL_data_POP_UNDER_18']))


Kids in UCVs

2010  21895
2020  27108


In [16]:
print('\nKids in entire city\n')
print('2010 ', sum(pop['F2010_PL_data_POP_UNDER_18']))
print('2020 ', sum(pop['F2020_PL_data_POP_UNDER_18']))


Kids in entire city

2010  93513
2020  106841


In [17]:
print("Fractional change in CHILD population in UCVs",
      round((sum(ucvpop['F2020_PL_data_POP_UNDER_18']) - 
             sum(ucvpop['F2010_PL_data_POP_UNDER_18']))/sum(ucvpop['F2010_PL_data_POP_UNDER_18']),
            3))
print()
print("Fractional change in CHILD population in entire city",
      round((sum(pop['F2020_PL_data_POP_UNDER_18']) - 
             sum(pop['F2010_PL_data_POP_UNDER_18']))/sum(pop['F2010_PL_data_POP_UNDER_18'])
            ,3), '\n')


Fractional change in CHILD population in UCVs 0.238

Fractional change in CHILD population in entire city 0.143 



In [18]:
print("Fractional change in ADULT population in UCVs", 
      round((sum(ucvpop['F2020_PL_data_POP_OVER_18']) - 
             sum(ucvpop['F2010_PL_data_POP_OVER_18']))/sum(ucvpop['F2010_PL_data_POP_OVER_18'])
            ,3))
print()
print("Fractional change in ADULT population in entire city",
      round((sum(pop['F2020_PL_data_POP_OVER_18']) - 
             sum(pop['F2010_PL_data_POP_OVER_18']))/sum(pop['F2010_PL_data_POP_OVER_18'])
            ,3),'\n')

Fractional change in ADULT population in UCVs 0.412

Fractional change in ADULT population in entire city 0.223 



In [19]:
# growth 2010-2020
print('Child population growth from 2010 to 2020')
print('UCV:     ', (sum(ucvpop['F2020_PL_data_POP_UNDER_18'])
                    - sum(ucvpop['F2010_PL_data_POP_UNDER_18'])))
print('Citywide:', (sum(pop['F2020_PL_data_POP_UNDER_18'])
                    - sum(pop['F2010_PL_data_POP_UNDER_18'])))
print()
print('Child population growth rate from 2010 to 2020')
print('UCV:     ', round((sum(ucvpop['F2020_PL_data_POP_UNDER_18'])
                          - sum(ucvpop['F2010_PL_data_POP_UNDER_18']))
                         / sum(ucvpop['F2010_PL_data_POP_UNDER_18']),3))
print('Citywide:', round((sum(pop['F2020_PL_data_POP_UNDER_18'])
                          - sum(pop['F2010_PL_data_POP_UNDER_18']))
                         / sum(pop['F2010_PL_data_POP_UNDER_18']),3))

Child population growth from 2010 to 2020
UCV:      5213
Citywide: 13328

Child population growth rate from 2010 to 2020
UCV:      0.238
Citywide: 0.143


From 2010 to 2020 the city as a whole saw an increase of 14.3% in the population of children.  Urban centers and villages as a whole saw a 23.8% rise in the child population over the same time period

Both UCVs and the city as a whole saw a larger fractional change in the adult population than in the child population.  The UCV adult population grew by 41.2% while the total adult population grew by 22.3% from 2010 to 2020.

In [20]:
ucvkids = ucvpop[['NEIGH_NAME','NEIGH_NO', 'AREA_ACRES']+[col for col in ucvpop.columns if 'UNDER' in col]]
kids = pop[['NEIGH_NAME','NEIGH_NO', 'AREA_ACRES']+[col for col in pop.columns if 'UNDER' in col]]

ucvadults = ucvpop[['NEIGH_NAME','NEIGH_NO', 'AREA_ACRES']+[col for col in ucvpop.columns if 'OVER' in col]]
adults = pop[['NEIGH_NAME','NEIGH_NO', 'AREA_ACRES']+[col for col in pop.columns if 'OVER' in col]]

In [21]:
ucvkids.columns = ['neighborhood','neighborhood_no','area','1990','2000','2010','2020']

In [22]:
ucvkids = ucvkids.copy()

In [23]:
ucvkids[ucvkids['neighborhood']=='Uptown']

neighborhood  neighborhood_no        area  1990  2000  2010  2020
OBJECTID                                                                  
20             Uptown              412  333.000698   172   147   222   596

In [24]:

ucvkids['change'] = ucvkids['2020']-ucvkids['2010']

In [25]:
ucvkids['frac_change'] = (ucvkids['2020']-ucvkids['2010'])/ucvkids['2010']

In [26]:
ucvkids['density'] = ucvkids['2020']/ucvkids['area']

In [27]:
ucvkids.sort_values('frac_change', ascending=False)

neighborhood  neighborhood_no         area  1990  2000  \
OBJECTID                                                                        
7                  South Lake Union              312   339.886927    80   122   
17        Ballard-Interbay-Northend              501   931.972864   107    66   
20                           Uptown              412   333.000698   172   147   
19                         Downtown              413   953.281747   276   755   
8               University District              401   751.963878   592   559   
12                 Greater Duwamish              502  4953.281793   148   213   
28                         Eastlake              107   199.255956   142   116   
14                       Green Lake              207   107.444638   103    94   
2                           Fremont              302   214.511083   266   171   
23              Bitter Lake Village              114   363.779566   379   396   
3             West Seattle Junction              205   270.417026   413   434   
4                     Columbia City              111   334.543340  1335  1423   
1                  Upper Queen Anne              106    52.658720   125   126   
6                         Roosevelt              202   170.211776   295   281   
30                          Ballard              303   495.476305   951   892   
9                        Crown Hill              103   270.988754   579   644   
11                        Lake City              201   142.267043   253   394   
29                      Wallingford              203   258.005613   448   477   
21                    Rainier Beach              204   346.143874  1159  1193   
27                  Morgan Junction              206   113.309377   191   243   
18          Greenwood-Phinney Ridge              102    94.157645   260   355   
25                          Admiral              113    98.248351   145   159   
13                   Madison-Miller              131   145.323420   480   334   
16                North Beacon Hill              110   266.876291  1001   992   
5           First Hill/Capitol Hill              414   916.221801  1729  1881   
22             23rd & Union-Jackson              109   625.266413  2036  2103   
15                        Northgate              301   412.291123   439   685   
26            Aurora-Licton Springs              101   326.958599   745   736   
24                         Mt Baker              305   491.018058   931  1016   
32                          Othello              123   498.963241  2338  2088   
10           Westwood-Highland Park              127   275.062196   970  1091   
31                       South Park              128   263.798620   638   855   

          2010  2020  change  frac_change   density  
OBJECTID                                             
7          158   549     391     2.474684  1.615243  
17          67   205     138     2.059701  0.219963  
20         222   596     374     1.684685  1.789786  
19         733  1607     874     1.192360  1.685756  
8          452   936     484     1.070796  1.244741  
12         135   245     110     0.814815  0.049462  
28         202   340     138     0.683168  1.706348  
14         163   265     102     0.625767  2.466386  
2          215   339     124     0.576744  1.580338  
23         597   929     332     0.556114  2.553744  
3          438   677     239     0.545662  2.503541  
4          968  1478     510     0.526860  4.417963  
1          136   201      65     0.477941  3.817032  
6          333   465     132     0.396396  2.731891  
30         964  1343     379     0.393154  2.710523  
9          747   927     180     0.240964  3.420806  
11         507   622     115     0.226824  4.372060  
29         701   839     138     0.196862  3.251867  
21        1087  1280     193     0.177553  3.697884  
27         268   294      26     0.097015  2.594666  
18         374   410      36     0.096257  4.354399  
25         198   215      

In [28]:
ucvkids.sort_values('2020', ascending=False).head(10)

neighborhood  neighborhood_no        area  1990  2000  \
OBJECTID                                                                     
32                        Othello              123  498.963241  2338  2088   
22           23rd & Union-Jackson              109  625.266413  2036  2103   
5         First Hill/Capitol Hill              414  916.221801  1729  1881   
19                       Downtown              413  953.281747   276   755   
4                   Columbia City              111  334.543340  1335  1423   
30                        Ballard              303  495.476305   951   892   
21                  Rainier Beach              204  346.143874  1159  1193   
24                       Mt Baker              305  491.018058   931  1016   
10         Westwood-Highland Park              127  275.062196   970  1091   
8             University District              401  751.963878   592   559   

          2010  2020  change  frac_change   density  
OBJECTID                                             
32        2759  2817      58     0.021022  5.645706  
22        1813  1902      89     0.049090  3.041903  
5         1786  1894     108     0.060470  2.067185  
19         733  1607     874     1.192360  1.685756  
4          968  1478     510     0.526860  4.417963  
30         964  1343     379     0.393154  2.710523  
21        1087  1280     193     0.177553  3.697884  
24        1046  1072      26     0.024857  2.183219  
10        1093   947    -146    -0.133577  3.442858  
8          452   936     484     1.070796  1.244741

In [29]:
ucvkids.sort_values('frac_change', ascending=False).head(10)

neighborhood  neighborhood_no         area  1990  2000  \
OBJECTID                                                                        
7                  South Lake Union              312   339.886927    80   122   
17        Ballard-Interbay-Northend              501   931.972864   107    66   
20                           Uptown              412   333.000698   172   147   
19                         Downtown              413   953.281747   276   755   
8               University District              401   751.963878   592   559   
12                 Greater Duwamish              502  4953.281793   148   213   
28                         Eastlake              107   199.255956   142   116   
14                       Green Lake              207   107.444638   103    94   
2                           Fremont              302   214.511083   266   171   
23              Bitter Lake Village              114   363.779566   379   396   

          2010  2020  change  frac_change   density  
OBJECTID                                             
7          158   549     391     2.474684  1.615243  
17          67   205     138     2.059701  0.219963  
20         222   596     374     1.684685  1.789786  
19         733  1607     874     1.192360  1.685756  
8          452   936     484     1.070796  1.244741  
12         135   245     110     0.814815  0.049462  
28         202   340     138     0.683168  1.706348  
14         163   265     102     0.625767  2.466386  
2          215   339     124     0.576744  1.580338  
23         597   929     332     0.556114  2.553744

In [30]:
ucvkids.sort_values('change', ascending=False).head(10)

neighborhood  neighborhood_no        area  1990  2000  \
OBJECTID                                                                   
19                     Downtown              413  953.281747   276   755   
4                 Columbia City              111  334.543340  1335  1423   
8           University District              401  751.963878   592   559   
7              South Lake Union              312  339.886927    80   122   
30                      Ballard              303  495.476305   951   892   
20                       Uptown              412  333.000698   172   147   
23          Bitter Lake Village              114  363.779566   379   396   
3         West Seattle Junction              205  270.417026   413   434   
21                Rainier Beach              204  346.143874  1159  1193   
9                    Crown Hill              103  270.988754   579   644   

          2010  2020  change  frac_change   density  
OBJECTID                                             
19         733  1607     874     1.192360  1.685756  
4          968  1478     510     0.526860  4.417963  
8          452   936     484     1.070796  1.244741  
7          158   549     391     2.474684  1.615243  
30         964  1343     379     0.393154  2.710523  
20         222   596     374     1.684685  1.789786  
23         597   929     332     0.556114  2.553744  
3          438   677     239     0.545662  2.503541  
21        1087  1280     193     0.177553  3.697884  
9          747   927     180     0.240964  3.420806

In [31]:
ucvkids.sort_values('density', ascending = False)

neighborhood  neighborhood_no         area  1990  2000  \
OBJECTID                                                                        
32                          Othello              123   498.963241  2338  2088   
4                     Columbia City              111   334.543340  1335  1423   
11                        Lake City              201   142.267043   253   394   
18          Greenwood-Phinney Ridge              102    94.157645   260   355   
1                  Upper Queen Anne              106    52.658720   125   126   
21                    Rainier Beach              204   346.143874  1159  1193   
10           Westwood-Highland Park              127   275.062196   970  1091   
9                        Crown Hill              103   270.988754   579   644   
13                   Madison-Miller              131   145.323420   480   334   
29                      Wallingford              203   258.005613   448   477   
16                North Beacon Hill              110   266.876291  1001   992   
22             23rd & Union-Jackson              109   625.266413  2036  2103   
6                         Roosevelt              202   170.211776   295   281   
30                          Ballard              303   495.476305   951   892   
31                       South Park              128   263.798620   638   855   
27                  Morgan Junction              206   113.309377   191   243   
23              Bitter Lake Village              114   363.779566   379   396   
26            Aurora-Licton Springs              101   326.958599   745   736   
3             West Seattle Junction              205   270.417026   413   434   
14                       Green Lake              207   107.444638   103    94   
25                          Admiral              113    98.248351   145   159   
24                         Mt Baker              305   491.018058   931  1016   
15                        Northgate              301   412.291123   439   685   
5           First Hill/Capitol Hill              414   916.221801  1729  1881   
20                           Uptown              412   333.000698   172   147   
28                         Eastlake              107   199.255956   142   116   
19                         Downtown              413   953.281747   276   755   
7                  South Lake Union              312   339.886927    80   122   
2                           Fremont              302   214.511083   266   171   
8               University District              401   751.963878   592   559   
17        Ballard-Interbay-Northend              501   931.972864   107    66   
12                 Greater Duwamish              502  4953.281793   148   213   

          2010  2020  change  frac_change   density  
OBJECTID                                             
32        2759  2817      58     0.021022  5.645706  
4          968  1478     510     0.526860  4.417963  
11         507   622     115     0.226824  4.372060  
18         374   410      36     0.096257  4.354399  
1          136   201      65     0.477941  3.817032  
21        1087  1280     193     0.177553  3.697884  
10        1093   947    -146    -0.133577  3.442858  
9          747   927     180     0.240964  3.420806  
13         450   482      32     0.071111  3.316740  
29         701   839     138     0.196862  3.251867  
16         803   857      54     0.067248  3.211226  
22        1813  1902      89     0.049090  3.041903  
6          333   465     132     0.396396  2.731891  
30         964  1343     379     0.393154  2.710523  
31         855   685    -170    -0.198830  2.596678  
27         268   294      26     0.097015  2.594666  
23         597   929     332     0.556114  2.553744  
26         799   828      29     0.036295  2.532431  
3          438   677     239     0.545662  2.503541  
14         163   265     102     0.625767  2.466386  
25         198   215      17     0.085859  2.188332  
24        1046  1072      

Ballard-Interbay-Northend and Greater Duwamish are density outliers.  Both of these districts are huge industrial areas of the city that are covered in the urban village development plan.  

Ballard-Interbay-Northend includes the former WA National Guard site recently given over to the city.  Redeveloment options for this area are currently being explored by a legislatively appointed [advisory council](https://www.commerce.wa.gov/about-us/research-services/interbay-public-development-advisory-committee/).  

Greater Duwamish is the area including the port of Seattle.  

For the purpose of current data we can exclude them from the residential population analysis as they are mainly industrial rather than mixed-use, residential/commerical areas.  



In [32]:
#dropping Ballard-Interbay-Northend and Greater Duwamish areas
ucvkids.drop(index = 12, inplace = True)
ucvkids.drop(index = 17, inplace = True)

In [33]:
# saving urban center kid data
ucvkids.to_csv('./data/seattle_ucv_kids_cleaned.csv')

Inconsistency in definition of Urban villages and urban centers leads to oddities in the density values.  Uptown, Downtown, South Lake Union and First Hill/Capitol Hill create a contiguous block.  That is, the entirety of central Seattle is in one of these UCs, even though there are vast variations in density inside each of the UCs.

The definition of UC boundaries for those outside of central Seattle are much tighter, owing to strict zoning laws and the fight against expansion of non-single family housing.  Development in these areas has been limited to directly along the main thoroughfare with the urban center/village defined as such.  This leads to misleadingly high densities for these neighborhoods.  

Furthermore if we look at each of the Urban centers outside of central Seattle, they are all in close proximity to at least one if not several elementary schools.  

Kids experiencing homeless are most often considered residents of the Downtown UC and are transported to Capitol Hill for school.  This is in part due to the concentration of shelters in the downtown area. (add citation) In 2017, 20% of students at the Lowell Elementary were experiencing homelessness.  This 



In [34]:
ucvkids['change2000to2010'] = ucvkids['2010']-ucvkids['2000']

In [35]:
ucvkids.sort_values('change2000to2010', ascending=False).head(10)

neighborhood  neighborhood_no        area  1990  2000  2010  \
OBJECTID                                                                       
32                    Othello              123  498.963241  2338  2088  2759   
29                Wallingford              203  258.005613   448   477   701   
23        Bitter Lake Village              114  363.779566   379   396   597   
15                  Northgate              301  412.291123   439   685   826   
13             Madison-Miller              131  145.323420   480   334   450   
11                  Lake City              201  142.267043   253   394   507   
9                  Crown Hill              103  270.988754   579   644   747   
28                   Eastlake              107  199.255956   142   116   202   
20                     Uptown              412  333.000698   172   147   222   
30                    Ballard              303  495.476305   951   892   964   

          2020  change  frac_change   density  change2000to2010  
OBJECTID                                                         
32        2817      58     0.021022  5.645706               671  
29         839     138     0.196862  3.251867               224  
23         929     332     0.556114  2.553744               201  
15         862      36     0.043584  2.090756               141  
13         482      32     0.071111  3.316740               116  
11         622     115     0.226824  4.372060               113  
9          927     180     0.240964  3.420806               103  
28         340     138     0.683168  1.706348                86  
20         596     374     1.684685  1.789786                75  
30        1343     379     0.393154  2.710523                72

In [36]:
ucvkids['perc_change2000to2010'] = ucvkids['change2000to2010']/ucvkids['2000']

In [37]:
ucvkids.sort_values('perc_change2000to2010',ascending=False).head(10)

neighborhood  neighborhood_no        area  1990  2000  2010  \
OBJECTID                                                                       
28                   Eastlake              107  199.255956   142   116   202   
14                 Green Lake              207  107.444638   103    94   163   
20                     Uptown              412  333.000698   172   147   222   
23        Bitter Lake Village              114  363.779566   379   396   597   
29                Wallingford              203  258.005613   448   477   701   
13             Madison-Miller              131  145.323420   480   334   450   
32                    Othello              123  498.963241  2338  2088  2759   
7            South Lake Union              312  339.886927    80   122   158   
11                  Lake City              201  142.267043   253   394   507   
2                     Fremont              302  214.511083   266   171   215   

          2020  change  frac_change   density  change2000to2010  \
OBJECTID                                                          
28         340     138     0.683168  1.706348                86   
14         265     102     0.625767  2.466386                69   
20         596     374     1.684685  1.789786                75   
23         929     332     0.556114  2.553744               201   
29         839     138     0.196862  3.251867               224   
13         482      32     0.071111  3.316740               116   
32        2817      58     0.021022  5.645706               671   
7          549     391     2.474684  1.615243                36   
11         622     115     0.226824  4.372060               113   
2          339     124     0.576744  1.580338                44   

          perc_change2000to2010  
OBJECTID                         
28                     0.741379  
14                     0.734043  
20                     0.510204  
23                     0.507576  
29                     0.469602  
13                     0.347305  
32                     0.321360  
7                      0.295082  
11                     0.286802  
2                      0.257310

In [38]:
ucvkids.columns

Index(['neighborhood', 'neighborhood_no', 'area', '1990', '2000', '2010',
       '2020', 'change', 'frac_change', 'density', 'change2000to2010',
       'perc_change2000to2010'],
      dtype='object')

In [39]:
ucvkids.sort_values('2020', ascending=False).head()

neighborhood  neighborhood_no        area  1990  2000  \
OBJECTID                                                                     
32                        Othello              123  498.963241  2338  2088   
22           23rd & Union-Jackson              109  625.266413  2036  2103   
5         First Hill/Capitol Hill              414  916.221801  1729  1881   
19                       Downtown              413  953.281747   276   755   
4                   Columbia City              111  334.543340  1335  1423   

          2010  2020  change  frac_change   density  change2000to2010  \
OBJECTID                                                                
32        2759  2817      58     0.021022  5.645706               671   
22        1813  1902      89     0.049090  3.041903              -290   
5         1786  1894     108     0.060470  2.067185               -95   
19         733  1607     874     1.192360  1.685756               -22   
4          968  1478     510     0.526860  4.417963              -455   

          perc_change2000to2010  
OBJECTID                         
32                     0.321360  
22                    -0.137898  
5                     -0.050505  
19                    -0.029139  
4                     -0.319747

In [40]:
ucvkids.sort_values('change', ascending=False).head()

neighborhood  neighborhood_no        area  1990  2000  2010  \
OBJECTID                                                                       
19                   Downtown              413  953.281747   276   755   733   
4               Columbia City              111  334.543340  1335  1423   968   
8         University District              401  751.963878   592   559   452   
7            South Lake Union              312  339.886927    80   122   158   
30                    Ballard              303  495.476305   951   892   964   

          2020  change  frac_change   density  change2000to2010  \
OBJECTID                                                          
19        1607     874     1.192360  1.685756               -22   
4         1478     510     0.526860  4.417963              -455   
8          936     484     1.070796  1.244741              -107   
7          549     391     2.474684  1.615243                36   
30        1343     379     0.393154  2.710523                72   

          perc_change2000to2010  
OBJECTID                         
19                    -0.029139  
4                     -0.319747  
8                     -0.191413  
7                      0.295082  
30                     0.080717

In [41]:
ucvkids['frac2020'] = ucvkids['2020']/np.sum(pop['F2020_PL_data_POP_UNDER_18'])

In [42]:
ucvkids.sort_values('frac2020', ascending=False).head(10)

neighborhood  neighborhood_no        area  1990  2000  \
OBJECTID                                                                     
32                        Othello              123  498.963241  2338  2088   
22           23rd & Union-Jackson              109  625.266413  2036  2103   
5         First Hill/Capitol Hill              414  916.221801  1729  1881   
19                       Downtown              413  953.281747   276   755   
4                   Columbia City              111  334.543340  1335  1423   
30                        Ballard              303  495.476305   951   892   
21                  Rainier Beach              204  346.143874  1159  1193   
24                       Mt Baker              305  491.018058   931  1016   
10         Westwood-Highland Park              127  275.062196   970  1091   
8             University District              401  751.963878   592   559   

          2010  2020  change  frac_change   density  change2000to2010  \
OBJECTID                                                                
32        2759  2817      58     0.021022  5.645706               671   
22        1813  1902      89     0.049090  3.041903              -290   
5         1786  1894     108     0.060470  2.067185               -95   
19         733  1607     874     1.192360  1.685756               -22   
4          968  1478     510     0.526860  4.417963              -455   
30         964  1343     379     0.393154  2.710523                72   
21        1087  1280     193     0.177553  3.697884              -106   
24        1046  1072      26     0.024857  2.183219                30   
10        1093   947    -146    -0.133577  3.442858                 2   
8          452   936     484     1.070796  1.244741              -107   

          perc_change2000to2010  frac2020  
OBJECTID                                   
32                     0.321360  0.026366  
22                    -0.137898  0.017802  
5                     -0.050505  0.017727  
19                    -0.029139  0.015041  
4                     -0.319747  0.013834  
30                     0.080717  0.012570  
21                    -0.088852  0.011980  
24                     0.029528  0.010034  
10                     0.001833  0.008864  
8                     -0.191413  0.008761

In [43]:
print('2020: ',sum(pop['F2020_PL_data_POP_UNDER_18']))
print('change: ',sum(pop['F2020_PL_data_POP_UNDER_18']) - sum(pop['F2010_PL_data_POP_UNDER_18']))

2020:  106841
change:  13328


In [44]:
print('2020: ',sum(ucvkids['2020']))
print('change: ',sum(ucvkids['2020']) - sum(ucvkids['2010']))

2020:  26658
change:  4965


In [45]:
(sum(ucvkids['2020']) - sum(ucvkids['2010']))/(sum(pop['F2020_PL_data_POP_UNDER_18']) - sum(pop['F2010_PL_data_POP_UNDER_18']))

0.3725240096038415

37% of the growth in child population was in UCVs.  



In [46]:
downtown_neighborhoods = ['Downtown','Uptown','Capitol Hill']

In [47]:
corekids = ucvkids[(ucvkids['neighborhood'] =='Downtown') | (ucvkids['neighborhood'] == 'Uptown') | (ucvkids['neighborhood']=='First Hill/Capitol Hill')]

In [48]:
corekids

neighborhood  neighborhood_no        area  1990  2000  \
OBJECTID                                                                     
5         First Hill/Capitol Hill              414  916.221801  1729  1881   
19                       Downtown              413  953.281747   276   755   
20                         Uptown              412  333.000698   172   147   

          2010  2020  change  frac_change   density  change2000to2010  \
OBJECTID                                                                
5         1786  1894     108     0.060470  2.067185               -95   
19         733  1607     874     1.192360  1.685756               -22   
20         222   596     374     1.684685  1.789786                75   

          perc_change2000to2010  frac2020  
OBJECTID                                   
5                     -0.050505  0.017727  
19                    -0.029139  0.015041  
20                     0.510204  0.005578

In [49]:
sum(corekids['2020'])

4097

In [50]:
sum(corekids['2010'])

2741

In [51]:
ucvkids.sort_values('change',ascending=False).head(10)

neighborhood  neighborhood_no        area  1990  2000  \
OBJECTID                                                                   
19                     Downtown              413  953.281747   276   755   
4                 Columbia City              111  334.543340  1335  1423   
8           University District              401  751.963878   592   559   
7              South Lake Union              312  339.886927    80   122   
30                      Ballard              303  495.476305   951   892   
20                       Uptown              412  333.000698   172   147   
23          Bitter Lake Village              114  363.779566   379   396   
3         West Seattle Junction              205  270.417026   413   434   
21                Rainier Beach              204  346.143874  1159  1193   
9                    Crown Hill              103  270.988754   579   644   

          2010  2020  change  frac_change   density  change2000to2010  \
OBJECTID                                                                
19         733  1607     874     1.192360  1.685756               -22   
4          968  1478     510     0.526860  4.417963              -455   
8          452   936     484     1.070796  1.244741              -107   
7          158   549     391     2.474684  1.615243                36   
30         964  1343     379     0.393154  2.710523                72   
20         222   596     374     1.684685  1.789786                75   
23         597   929     332     0.556114  2.553744               201   
3          438   677     239     0.545662  2.503541                 4   
21        1087  1280     193     0.177553  3.697884              -106   
9          747   927     180     0.240964  3.420806               103   

          perc_change2000to2010  frac2020  
OBJECTID                                   
19                    -0.029139  0.015041  
4                     -0.319747  0.013834  
8                     -0.191413  0.008761  
7                      0.295082  0.005138  
30                     0.080717  0.012570  
20                     0.510204  0.005578  
23                     0.507576  0.008695  
3                      0.009217  0.006337  
21                    -0.088852  0.011980  
9                      0.159938  0.008676

## School Enrollment Data

Looking over the last 20 years, Seattle Public Schools serves approximately 50% of the children in the city.  Over the last 5 years the average enrollment across the district has been 53,300 students (K-12).  The pandemic saw a decrease in enrollment in 2020 with many families delaying kindergarten and choosing homeschooling or delayed start.  Pre-pandemic data showed steady enrollment in kindergarten year over year.  During the same time period there were minor declines in 1st-5th enrollment of abpproximately 1% year over year, an average increase of 1.6% in 6-8th enrollment, and an average increase in 9-12 enrollment of 1.2% per year.  All told the district serves roughly half of the total children in the city.  The remainder are either too young for public schools (Seattle has a separate preschool program) or are served by private schools or homeschooling.  

## Housing Data and Projections

The city monitors permits for new housing and projects housing development across the city.  

# To Do

- work on getting enrollment data in the correct form
- get adult population data in addition to kid data for the test.py dash
- Look at next few years of housing applications and predictions
- Housing data from 2010 and predicton for future
- Nearest schools to each "center"

Longer term
- get data on moving rates?  get data on birth rates?
- get data on kid ages?
- fraction of Lowell that is experiencing homelessness (by year)
- Ask Vivian about the data used for enrollment forecasting
- Ask Andrew Lewis about what information would be helpful.


## Other Questions
- why does district 4 sneak over to Ballard?  Is that so that it would include a high school?  The district lines do not respect school assignments.  Why is that?
- 


## Prepare Dashboard Data
To get the data ready for the dashboard we reorganize the dataframe to have column for year, indicator and value.  Each record has a neighborhood, neighborhood_no, and area as well.  


Indicators are total population for the year, kid population for the year, 10 year growth in total population, 10 year growth in kid population, 10 year fractional growth in total population, 10 year fractional growth in kid population, density of total population, and density of kids.  

In [186]:
# Reorganizing total population data for the dashboard

# break down into years and compute growth
# start with a copy of the kid dataframe
df = ucvpop[['NEIGH_NO', 'NEIGH_NAME','AREA_ACRES', 'F1990_PL_data_TOT_POP','F2000_PL_data_TOT_POP',
      'F2010_PL_data_TOT_POP','F2020_PL_data_TOT_POP']].copy() 

# drop the two industrial districts 
df.drop(index = 12, inplace = True) 
df.drop(index = 17, inplace = True)

# rename columns to be readable
df.columns = ['neighborhood_no','neighborhood','area','1990','2000','2010','2020']

# break it down by year
df1990 = df[['neighborhood', 'neighborhood_no','area','1990']].copy()
df2000 = df[['neighborhood', 'neighborhood_no','area','2000']].copy()
df2010 = df[['neighborhood', 'neighborhood_no','area','2010']].copy()
df2020 = df[['neighborhood', 'neighborhood_no','area','2020']].copy()

# add a column for year
df1990['Year'] = 1990
df2000['Year'] = 2000
df2010['Year'] = 2010
df2020['Year'] = 2020

# rename columns to be readable
df1990.columns = ['neighborhood', 'neighborhood_no', 'area', 'total population', 'Year']
df2000.columns = ['neighborhood', 'neighborhood_no', 'area', 'total population', 'Year']
df2010.columns = ['neighborhood', 'neighborhood_no', 'area', 'total population', 'Year']
df2020.columns = ['neighborhood', 'neighborhood_no', 'area', 'total population', 'Year']

# compute growth values
df2020['10_year_growth_all'] = df2020['total population'] - df2010['total population']
df2010['10_year_growth_all'] = df2010['total population'] - df2000['total population']
df2000['10_year_growth_all'] = df2000['total population'] - df1990['total population']

df2020['10_year_frac_growth_all'] = (df2020['total population'] - df2010['total population'])/df2010['total population']
df2010['10_year_frac_growth_all'] = (df2010['total population'] - df2000['total population'])/df2000['total population']
df2000['10_year_frac_growth_all'] = (df2000['total population'] - df1990['total population'])/df1990['total population']

df2020['total density'] = (df2020['total population'])/df2010['area']
df2010['total density'] = (df2010['total population'])/df2000['area']
df2000['total density'] = (df2000['total population'])/df1990['area']
df1990['total density'] = (df1990['total population'])/df1990['area']

# Reorganize the dataframes: turn the individual columns into rows 
#    we have a column for `Indicator Name` and a column for `Value`

# start with a datframe with all years combined
fullpopdf = pd.concat([df1990,df2000,df2010,df2020])
#fullpopdf['density'] = fullpopdf['total population']/fullpopdf['area']

# identify the columns we will need
# basic columns appear in all records
basiccols = ['neighborhood_no', 'neighborhood','area']
indicators = ['total population','total density','10_year_growth_all','10_year_frac_growth_all']


dfs = []#empty list for accumulating dataframes to concatenate

# get an individual dataframe for each indicator
for ind in indicators:
    subdf = fullpopdf.copy()
    subdf['Indicator Name'] = ind    
    subdf['Value'] = subdf[ind]
    subdf.drop(columns=ind, inplace = True)
    dfs.append(subdf)

    
# put all the dataframes together and drop indicator columns
allagesdf = pd.concat(dfs)
allagesdf.drop(columns = ['10_year_growth_all','10_year_frac_growth_all','total density','total population'], inplace = True)

# save the dataframe to a file
allagesdf.to_csv('./data/dash-data-allages.csv')

In [187]:
# Now do the same data reorginzation for the kid data

# break down into years and compute growth
df = ucvkids.copy() #start with a copy of the kid dataframe

# break it down by year
df1990 = df[['neighborhood', 'neighborhood_no','area','1990']].copy()
df2000 = df[['neighborhood', 'neighborhood_no','area','2000']].copy()
df2010 = df[['neighborhood', 'neighborhood_no','area','2010']].copy()
df2020 = df[['neighborhood', 'neighborhood_no','area','2020']].copy()

# add a column for year
df1990['Year'] = 1990
df2000['Year'] = 2000
df2010['Year'] = 2010
df2020['Year'] = 2020

# change column names to be readable
df1990.columns = ['neighborhood', 'neighborhood_no', 'area', 'population_kids', 'Year']
df2000.columns = ['neighborhood', 'neighborhood_no', 'area', 'population_kids', 'Year']
df2010.columns = ['neighborhood', 'neighborhood_no', 'area', 'population_kids', 'Year']
df2020.columns = ['neighborhood', 'neighborhood_no', 'area', 'population_kids', 'Year']

# compute growth values
df2020['10_year_growth_kids'] = df2020['population_kids'] - df2010['population_kids']
df2010['10_year_growth_kids'] = df2010['population_kids'] - df2000['population_kids']
df2000['10_year_growth_kids'] = df2000['population_kids'] - df1990['population_kids']

df2020['10_year_frac_growth_kids'] = (df2020['population_kids'] - df2010['population_kids'])/df2010['population_kids']
df2010['10_year_frac_growth_kids'] = (df2010['population_kids'] - df2000['population_kids'])/df2000['population_kids']
df2000['10_year_frac_growth_kids'] = (df2000['population_kids'] - df1990['population_kids'])/df1990['population_kids']

df2020['density_kids'] = (df2020['population_kids'])/df2010['area']
df2010['density_kids'] = (df2010['population_kids'])/df2000['area']
df2000['density_kids'] = (df2000['population_kids'])/df1990['area']
df1990['density_kids'] = (df1990['population_kids'])/df1990['area']

# Reorganize the dataframes: turn the individual columns into rows 
#    we have a column for `Indicator Name` and a column for `Value`

# start with a datframe with all years combined
fulldf = pd.concat([df1990,df2000,df2010,df2020])

# identify the columns we will need
# basic columns appear in all records
basiccols = ['neighborhood_no', 'neighborhood','area']
indicators = ['population_kids','density_kids','10_year_growth_kids','10_year_frac_growth_kids']

dfs = [] #empty list for accumulating dataframes to concatenate

# get an individual dataframe for each indicator
for ind in indicators:
    subdf = fulldf.copy()
    subdf['Indicator Name'] = ind    
    subdf['Value'] = subdf[ind]
    dfs.append(subdf)
    
kidsdf = pd.concat(dfs) #put dfs for all indicators together
kidsdf.drop(columns = indicators, inplace = True) #drop the old columns

#save dataframe
kidsdf.to_csv('./data/dash-data-kids.csv')

In [197]:
# Save the combined kid and total population data to a dataframe 
combineddf = pd.concat([kidsdf,allagesdf])
combineddf.to_csv('./data/dash-data-combined.csv')

## Dashboard To Do:
- Fix hover name not being the neighborhood anymore....what the what????
- what is going on with som indicators (10_year_growth_all, 10 year frac growth all, density
) not showing up on time series plot?
- Fix density is not by year (but density_kids may be...)
- rename variables to meaningful things?  Don't really need underscores here...
- make it pretty

In [189]:
combineddf[combineddf['Indicator Name']=='total population']

neighborhood  neighborhood_no        area  Year  \
OBJECTID                                                               
1                Upper Queen Anne              106   52.658720  1990   
2                         Fremont              302  214.511083  1990   
3           West Seattle Junction              205  270.417026  1990   
4                   Columbia City              111  334.543340  1990   
5         First Hill/Capitol Hill              414  916.221801  1990   
...                           ...              ...         ...   ...   
28                       Eastlake              107  199.255956  2020   
29                    Wallingford              203  258.005613  2020   
30                        Ballard              303  495.476305  2020   
31                     South Park              128  263.798620  2020   
32                        Othello              123  498.963241  2020   

            Indicator Name    Value  
OBJECTID                             
1         total population   1921.0  
2         total population   3153.0  
3         total population   3627.0  
4         total population   4281.0  
5         total population  28940.0  
...                    ...      ...  
28        total population   6100.0  
29        total population   6506.0  
30        total population  17707.0  
31        total population   3242.0  
32        total population  11257.0  

[120 rows x 6 columns]

In [190]:
combineddf.columns

Index(['neighborhood', 'neighborhood_no', 'area', 'Year', 'Indicator Name',
       'Value'],
      dtype='object')

In [191]:
combineddf['Indicator Name'].unique()

array(['population_kids', 'density_kids', '10_year_growth_kids',
       '10_year_frac_growth_kids', 'total population', 'total density',
       '10_year_growth_all', '10_year_frac_growth_all'], dtype=object)

In [193]:
combineddf[(combineddf['Year']==2020) & (combineddf['Indicator Name']=='total density')]

neighborhood  neighborhood_no        area  Year  \
OBJECTID                                                               
1                Upper Queen Anne              106   52.658720  2020   
2                         Fremont              302  214.511083  2020   
3           West Seattle Junction              205  270.417026  2020   
4                   Columbia City              111  334.543340  2020   
5         First Hill/Capitol Hill              414  916.221801  2020   
6                       Roosevelt              202  170.211776  2020   
7                South Lake Union              312  339.886927  2020   
8             University District              401  751.963878  2020   
9                      Crown Hill              103  270.988754  2020   
10         Westwood-Highland Park              127  275.062196  2020   
11                      Lake City              201  142.267043  2020   
13                 Madison-Miller              131  145.323420  2020   
14                     Green Lake              207  107.444638  2020   
15                      Northgate              301  412.291123  2020   
16              North Beacon Hill              110  266.876291  2020   
18        Greenwood-Phinney Ridge              102   94.157645  2020   
19                       Downtown              413  953.281747  2020   
20                         Uptown              412  333.000698  2020   
21                  Rainier Beach              204  346.143874  2020   
22           23rd & Union-Jackson              109  625.266413  2020   
23            Bitter Lake Village              114  363.779566  2020   
24                       Mt Baker              305  491.018058  2020   
25                        Admiral              113   98.248351  2020   
26          Aurora-Licton Springs              101  326.958599  2020   
27                Morgan Junction              206  113.309377  2020   
28                       Eastlake              107  199.255956  2020   
29                    Wallingford              203  258.005613  2020   
30                        Ballard              303  495.476305  2020   
31                     South Park              128  263.798620  2020   
32                        Othello              123  498.963241  2020   

         Indicator Name      Value  
OBJECTID                            
1         total density  49.450499  
2         total density  27.229362  
3         total density  33.873607  
4         total density  22.899873  
5         total density  53.670410  
6         total density  27.483410  
7         total density  37.968509  
8         total density  39.745260  
9         total density  18.056100  
10        total density  17.272457  
11        total density  35.714526  
13        total density  38.610432  
14        total density  36.614205  
15        total density  19.282491  
16        total density  21.538069  
18        total density  37.086739  
19        total density  43.927202  
20        total density  35.555481  
21        total density  13.953735  
22        total density  21.766722  
23        total density  16.768396  
24        total density  13.801936  
25        total density  20.132653  
26        total density  21.605182  
27        total density  20.880884  
28        total density  30.613890  
29        total density  25.216506  
30        total density  35.737330  
31        total density  12.289678  
32        total density  22.560780

In [196]:
combineddf['neighborhood'].unique()

array(['Upper Queen Anne', 'Fremont', 'West Seattle Junction',
       'Columbia City', 'First Hill/Capitol Hill', 'Roosevelt',
       'South Lake Union', 'University District', 'Crown Hill',
       'Westwood-Highland Park', 'Lake City', 'Madison-Miller',
       'Green Lake', 'Northgate', 'North Beacon Hill',
       'Greenwood-Phinney Ridge', 'Downtown', 'Uptown', 'Rainier Beach',
       '23rd & Union-Jackson', 'Bitter Lake Village', 'Mt Baker',
       'Admiral', 'Aurora-Licton Springs', 'Morgan Junction', 'Eastlake',
       'Wallingford', 'Ballard', 'South Park', 'Othello'], dtype=object)